NewTutorials/Tutorial/PlayerViewControllerWrapper.swift

```diff
@@ -8,42 +8,51 @@ import PlayerControls
 class PlayerViewControllerWrapper: UIViewController {
     struct Props {
         var isLoading = false
+        var player: Future<Result<Player>>?
+        var controlsViewController: ContentControlsViewController = DefaultControlsViewController()
     }
     
     var props = Props() {
         didSet {
             view.setNeedsLayout()
         }
     }
-    
+    @IBOutlet weak private var activityIndicatorView: UIActivityIndicatorView!
     private var playerViewController: PlayerViewController? {
         return childViewControllers.first as? PlayerViewController
     }
-    private let activityIndicatorView: UIActivityIndicatorView = {
-        let activityIndicatorView = UIActivityIndicatorView()
-        activityIndicatorView.activityIndicatorViewStyle = .gray
-        activityIndicatorView.isHidden = true
-        return activityIndicatorView
-    }()
-    
-    func attach(player: Player) {
-        playerViewController?.player = player
-    }
-    
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        
-        playerViewController?.contentControlsViewController = DefaultControlsViewController()
-        activityIndicatorView.color = view.tintColor
-        navigationItem.rightBarButtonItem = UIBarButtonItem(customView: activityIndicatorView)
-    }
     
     override func viewWillLayoutSubviews() {
         super.viewWillLayoutSubviews()
         activityIndicatorView.isHidden = !props.isLoading
         props.isLoading ?
             activityIndicatorView.startAnimating() :
             activityIndicatorView.stopAnimating()
+        
+        playerViewController?.contentControlsViewController = props.controlsViewController
+        
+        guard !props.isLoading else { return }
+        guard self.playerViewController?.player == nil else { return }
+        guard let player = props.player else { return }
+        
+        func show(error: Error) {
+            let alert = UIAlertController(title: "Error",
+                                          message: "\(error)",
+                preferredStyle: .alert)
+            alert.addAction(.init(title: "OK",
+                                  style: .default,
+                                  handler: nil))
+            present(alert,
+                    animated: true,
+                    completion: nil)
+        }
+        
+        props.isLoading = true
+        player
```

Comment:
Why this code is here - it should be in `didSet` of `props`.
Here we have to update UI only from corresponding props.